scipy/stats/_stats_py.py

```diff
@@ -593,29 +605,50 @@ def mode(a, axis=0, nan_policy='propagate'):
     ...               [5, 3, 0, 5],
     ...               [4, 7, 5, 9]])
     >>> from scipy import stats
-    >>> stats.mode(a)
-    ModeResult(mode=array([3, 1, 0, 0]), count=array([1, 1, 1, 1]))
+    >>> stats.mode(a, keepdims=True)
+    ModeResult(mode=array([[3, 0, 6, 1]]), count=array([[4, 2, 2, 1]]))
```

Comment:
@mdhaber There's definitely something wrong here--when I was rebasing I was worried that the data in `a` was different vs. `main` I think, and that's related to the current `stats` refguide fail I believe.

Reply:
Yeah the input data changed in https://github.com/scipy/scipy/pull/16629. If we're not changing the input data, we don't want to change the output data. But due to https://github.com/scipy/scipy/pull/16429, the output should be be 2D.

```suggestion
    ModeResult(mode=array([[3, 1, 0, 0]]), count=array([[1, 1, 1, 1]]))
```
I think this is right.